apis/pipelines/v1alpha6/artifact.go

```diff
@@ -0,0 +1,89 @@
+package v1alpha6
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+type ArtifactLocator struct {
+	Component string `json:"-" yaml:"-"`
+	Artifact  string `json:"-" yaml:"-"`
+	Index     int    `json:"-" yaml:"-"`
+}
+
+func (ap *ArtifactLocator) String() string {
+	return fmt.Sprintf("%s:%s:%d", ap.Component, ap.Artifact, ap.Index)
+}
+
+const ArtifactPathPattern = `^([^\[\]:]+):([^\[\]:]+)(?::(\d*))?(?:\[([^\[\]:]+)\])?$`
```

Comment:
can be private?

Reply:
Can be but this was just straight copy/pasted.

Reply:
Think I would prefer to leave other changes to a minimum on this branch if thats ok and just leave this as is.